gammapy/datasets/simulate.py

```diff
@@ -70,25 +79,58 @@ def _evaluate_timevar_source(self, dataset, evaluator, time_axis=None):
 
         Returns
         -------
-        npred : `~gammapy.maps.Map`
+        npred : `~gammapy.maps.RegionNDMap`
             Npred map.
         """
-        energy_axis = MapAxis.from_edges(
-            dataset.geoms["geom"].axes["energy"].edges, name="energy_true"
+        energy_true = dataset.edisp.edisp_map.geom.axes["energy_true"]
+        energy_new = energy_true.upsample(10)
```

Comment:
We should avoid the hardcoded number here..

Reply:
I agree, but I'm not sure what's the best way to avoid it. Here we need to oversample the energy to have a precise estimate of `npred`, so how could we do it in a universal approach? Should we left this to the users? Any suggestion @adonath , @registerrier , @AtreyeeS ?

Reply:
I think we always said that the geometry and energy binning of of the `MapDataset` should be adapted to the required precision. So when in doubt users should choose a fine energy binning. I do not see how the behavior would be different compared to a "normal" spectral model.

Reply:
I understand your point and I agree when we talk about normal models. In this case however models are, in some sense, not common. They can be more complex than standard ones and users can do mistakes: for example spectra in different times bins may have very different shapes and it cannot be excluded that a (fine) energy binning is ok for one spectrum but not enough for the other. 
Upsampling the energies is a sort of advice that we give to the user...